src/mono/mono/utils/mono-threads-wasm.c

```diff
@@ -227,11 +195,11 @@ mono_threads_platform_get_stack_bounds (guint8 **staddr, size_t *stsize)
 		g_error ("%s: pthread_attr_destroy failed with \"%s\" (%d)", __func__, g_strerror (res), res);
 
 	if (*staddr == NULL) {
-		*staddr = (guint8*)wasm_get_stack_base ();
+		*staddr = (guint8*)get_wasm_stack_low ();
 		*stsize = wasm_get_stack_size ();
 	}
 #else
-	*staddr = (guint8*)wasm_get_stack_base ();
+	*staddr = (guint8*)get_wasm_stack_low ();
```

Comment:
quite likely, this is plain wrong for MT WASI. maybe rather assert instead ?

Reply:
because the shadow stack of the thread is not allocated by the LLVM but by the libc/pthread POSIX emulation. 

Eventually we could try to fix it for cases when pthreads are enabled on WASI+libc. But there is long way to go for that.

Reply:
Done

Reply:
Looks like we do reach this code so the assert gets triggered

Reply:
Is the `if (*staddr == NULL) *staddr = (guint8*)get_wasm_stack_low ();` fallback a few lines above for `__EMSCRIPTEN_PTHREADS__` is wrong too?

Reply:
Maybe it solves fallback for main thread ?

Reply:
or maybe we never actually hit it?

Reply:
added an assert to verify we never actually hit this for the main thread either.